### ISSUE TITLE
Add spec for BatchNormGradOp

### ISSUE THREAD

**Issue by sdasgup3** [openxla/stablehlo#345]
### Request description

To spec stablehlo.batch_norm_grad
